Chia Chu You iP

chuyouchia opened the pull request.

chuyouchia committed:
Duke Level 1: Greet, Echo & Exit

chuyouchia committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

chuyouchia committed:
Add the ability to mark tasks as done.

chuyouchia committed:
Add support for tracking three types of tasks: Todos, Deadlines and Events

chuyouchia committed:
Use the input/output redirection technique to semi-automate the testing of Duke.

chuyouchia committed:
Teach Duke to deal with errors such as incorrect inputs entered by the user.

chuyouchia committed:
Fixed bug in list command for level 5. Level 5 - Teach Duke to deal with errors such as incorrect inputs entered by the user.

chuyouchia committed:
Fixed Minor bug in level 5 due to missing white space in Task's getCurrentStatus method call. Level-5: Teach Duke to deal with errors such as incorrect inputs entered by the user.

chuyouchia committed:
Level-6: Add support for deleting tasks from the list.

chuyouchia committed:
Refactor Duke.java to improve readability and extensibility:

Original code contained UI and logic in the main method.

As a result, if there are any changes to be made to a specific scenario, problems are not isolated into individual methods or cases.

Simplified the driver method to be solely responsible for console input and dependent on other static methods.

By separating the responsibility into smaller chunks, it will improve debugging process if necessary.

Java Doc comments have been added to describe individual methods as well.

chuyouchia committed:
Add ability to parse date time format in command:

Original code was treating all date time formats simply as strings.

Add the function for the tasks to have stored date time. Command "list-due" added for filtering options on deadline and commitments.

chuyouchia committed:
Add ability to access file system & refactor commands into Command Handler.

Save the task list to hard disk so that past tasks recorded would not be lost when the program terminates.

chuyouchia committed:
Merge branch 'branch-Level-8' with master

# Conflicts:
#	src/main/java/com/jacob/Duke/Deadline.java
#	src/main/java/com/jacob/Duke/Duke.java
#	src/main/java/com/jacob/Duke/Event.java
#	src/main/java/com/jacob/Duke/Task.java

chuyouchia committed:
Refactor the code to extract out closely related code as classes & Place into Packages

Original code had repeated UI elements and combined Commands with Parser, which resulted in unnecessary repeated code between commands.

Extracted the following classes:
- Ui: deals with interactions with the user
- Renamed FileEditor to Storage: deals with loading tasks from the file and saving tasks in the file
- Parser: deals with making sense of the user command
- TaskList: contains the task list e.g., it has operations to add/delete tasks in the list
- Command: that implements a Command Interface with Execute and isBye methods.

Extracting the class allows for method reusing and modularity as codebase grows.

chuyouchia committed:
Add JUnit Testing and Fix UI bug for doneCommand.

JUnit enables automated testing for Test Driven Development.

Note:
-     implementation 'org.junit.jupiter:junit-jupiter-api:5.5.1'  is required to make sure that the import for @Test annotation does not fail
-     Set the System.out to a byteArrayOutput Stream so that it is possible to capture console output for testing purposes.

chuyouchia committed:
Fix file creation bug discovered during JAR file level

chuyouchia committed:
Add JavaDoc comments for all public methods in production code

chuyouchia committed:
Tweak all code to comply with given coding standard

chuyouchia committed:
Fix bug resulting in unhandled exception due to non-existent breakpoint

chuyouchia committed:
Add a find command to search tasks for specified string

chuyouchia committed:
Merge branch 'Branch-A-CodingStandard'

# Conflicts:
#	src/main/java/com/jacob/duke/DukeException.java
#	src/main/java/com/jacob/duke/TaskList.java
#	src/main/java/com/jacob/duke/command/ByeCommand.java
#	src/main/java/com/jacob/duke/command/DeadlineCommand.java
#	src/main/java/com/jacob/duke/command/DeleteCommand.java
#	src/main/java/com/jacob/duke/command/DoneCommand.java
#	src/main/java/com/jacob/duke/command/EventCommand.java
#	src/main/java/com/jacob/duke/command/PrintFilteredListCommand.java
#	src/main/java/com/jacob/duke/command/PrintListCommand.java
#	src/main/java/com/jacob/duke/task/Todo.java

chuyouchia committed:
Merge branch 'branch-Level-9'

Give users a way to find a task by searching for a keyword.

# Conflicts:
#	src/main/java/com/jacob/duke/Ui.java
#	src/main/java/com/jacob/duke/command/PrintFilteredListDateTimeCommand.java

yuxuanxc commented on src/main/java/com/jacob/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+
+    /**
+     * Constructor for Duke
+     * @param filePath contains file where task list is saved
```
There should be a punctuation (full-stop) behind each parameter description.
Remember to put an empty line between description and parameter section.

chuyouchia replied:
I see, thank you for pointing that out!

yuxuanxc commented on src/main/java/com/jacob/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+    private Ui ui;
+
+    /**
+     * Constructor for Duke
```
I think we need to include a full-stop at the back of the summary to indicate the end of the sentence, so that Javadoc will automatically place it in the method summary table(of your API)?

yuxuanxc commented on src/main/java/com/jacob/duke/Parser.java:
```diff
@@ -0,0 +1,58 @@
+package main.java.com.jacob.duke;
+
+import main.java.com.jacob.duke.command.*;
```
I think we still need to import every single class specifically instead of using a wildcard(*), even if all of the classes in that package is being used. I went to google about this a little but there are conflicting answers as to whether wildcards should be used or not, so maybe you can clarify it with your TA/the profs?

chuyouchia replied:
I actually did try that, but it seems like my IntelliJ automatically refactors the code to the * type import. Are you facing the same problems?

yuxuanxc replied:
I found this online! (source: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#exclude-from-auto-import)
Disable wildcard imports to always import single classes﻿:
1) In the Settings/Preferences dialog Ctrl+Alt+S, select Code Style | Java | Imports.
2) Make sure that the Use single class import option is enabled.
3) In the Class count to use import with ‘*’ and Names count to use static import with ‘*’ fields, specify values that definitely exceed the number of classes in a package and the number of names in a class (for example, 999).

chuyouchia replied:
Thank you, that works 👍

yuxuanxc commented on src/main/java/com/jacob/duke/Storage.java:
```diff
@@ -0,0 +1,228 @@
+import main.java.com.jacob.duke.task.Task;
+import main.java.com.jacob.duke.task.Todo;
+
+
```
There's a few extra spaces here.

chuyouchia replied:
I see, thank you for pointing that out!

yuxuanxc commented on src/main/java/com/jacob/duke/Storage.java:
```diff
@@ -0,0 +1,228 @@
+        java.nio.file.Path directoryPath = java.nio.file.Paths.get(current, parent);
+        boolean directoryExists = java.nio.file.Files.exists(directoryPath);
+
+
```
There's one extra space here, not sure if it is a violation but you might want to delete it so that your code will look more consistent throughout? There is one extra space at line 65 too

yuxuanxc commented on src/main/java/com/jacob/duke/Storage.java:
```diff
@@ -0,0 +1,228 @@
+            BufferedReader reader = new BufferedReader(new InputStreamReader(
+                    new DataInputStream(new FileInputStream(this.filename))));
+            int count = 0;
+            for (String line; (line = reader.readLine()) != null; ) {
```
Interesting way of using a for loop, I'm not very sure how it works but if (String) line is not updated then maybe you can use a while loop instead?

yuxuanxc commented on src/main/java/com/jacob/duke/Ui.java:
```diff
@@ -0,0 +1,150 @@
+     */
+    public void showNewTaskAdded(String taskDescription, List<Task> taskList) {
+        System.out.printf(
+                " Got it. I've added this task: \n   %s\n Now you have %d tasks in the list.\n", taskDescription, taskList.size());
```
Not a violation but you could follow the format of showTaskDeleted to improve readability!

yuxuanxc commented on src/main/java/com/jacob/duke/Ui.java:
```diff
@@ -0,0 +1,150 @@
+        System.out.printf(" Noted. I've removed this task:\n "
+                + "   %s\n"
+                + " Now you have %d tasks in the list.\n", taskDescription, taskList.size());
+
```
Random empty space here

yuxuanxc commented on src/main/java/com/jacob/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,31 @@
+package main.java.com.jacob.duke.command;
+import java.util.List;
```
Need to add a space above!

yuxuanxc commented on src/main/java/com/jacob/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,58 @@
+package main.java.com.jacob.duke.command;
+import java.util.List;
```
Need space above this line.

yuxuanxc commented on src/main/java/com/jacob/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,58 @@
+    public DeadlineCommand(String fullCommand) {
+        this.inputCommand = fullCommand;
+    }
+    /**
```
Need space above this line?

yuxuanxc commented on src/main/java/com/jacob/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,58 @@
+import main.java.com.jacob.duke.TaskList;
+import main.java.com.jacob.duke.Ui;
+
+import main.java.com.jacob.duke.task.Task;
```
Extra spaces in lines 8, 11 and 12

yuxuanxc commented on src/main/java/com/jacob/duke/command/PrintFilteredListDateTimeCommand.java:
```diff
@@ -0,0 +1,44 @@
+import main.java.com.jacob.duke.Ui;
+import main.java.com.jacob.duke.task.Task;
+
+
```
Why got so many spaces haha. This is not a violation so I don't want to keep spamming the same comment.

Just in case if you want to fix the extra spaces for the sake of aesthetics, here are some which I found in other files which I didn't comment:
TodoCommand.java - line 12
Event.java - line 3
UiTest.java - line 14

chuyouchia replied:
Thank you for pointing that out! Will refactor accordingly in my next release 👍

yuxuanxc reviewed (commented):
I think the main issue is with the javadoc comments, maybe you will need to spend some time fixing your current comments to adhere to the coding standard guidelines. Remember to put a full stop behind every summary, param, return etc and leave an empty line after the summary before you start writing the parameters in the future!  Sorry for spamming a lot of comments regarding extra spaces, I just thought that removing them would make your code look more consistent and organized. Other than that, the naming and layout all seems fine. Good job! :D

chuyouchia committed:
Add GUI using Java Fx Technology

Refactor UI and UI testing to support string validation

chuyouchia committed:
Add welcome message for duke

chuyouchia committed:
Add assertions to source code to document assumptions

chuyouchia committed:
Improve Code Quality by extracting and naming variables.

Add changes to adhere to JavaDocs Coding standards

chuyouchia committed:
Merge pull request #1 from chuyouchia/branch-A-Assertions

Add assertions to source code to document assumptions

chuyouchia committed:
Merge branch 'master' into branch-A-CodeQuality

chuyouchia committed:
Merge pull request #2 from chuyouchia/branch-A-CodeQuality

Merge Code Quality improvements by extracting and naming variables.

chuyouchia committed:
Update image file path

chuyouchia committed:
Add CI via workflows/gradle.yml

chuyouchia committed:
Update README.md

chuyouchia committed:
Update gradle.yml location

chuyouchia committed:
Fix gradlew build problem in gradle.yml

chuyouchia committed:
Update to remove checkstyle errors detected by CI

hopinxian commented on data/duke.txt:
```diff
@@ -0,0 +1,4 @@
+D,1,return book,2019-11-13 1800
+T,0,funny
+T,0,nani what is happening to me
+E,0,omomomo,2019-11-15 1900
```
May I suggest adding more line inputs with respect to the increasing levels? I find that maintaining this text assists greatly with testing.

chuyouchia replied:
Yup this make sense. I actually have separate test files and duke.txt is really just a dummy text file that I ignored after the start.

hopinxian commented on src/main/java/com/jacob/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+package main.java.com.jacob.duke;
```
May I suggest putting the Parser and Storage class within a IO package?

chuyouchia replied:
Makes sense, thanks for the suggestion!

hopinxian commented on src/main/java/com/jacob/duke/Storage.java:
```diff
@@ -0,0 +1,179 @@
+    }
+
+    /**
+     * Replaces a specific line of text in the string buffer read
+     * @param lineToEdit text to be replaced
```
I believe within the coding standard, it is stated there should be a line break between the description and parameter block. Perhaps you check to confirm? This goes for other places within this project as well.

chuyouchia replied:
Yup thanks, have rectified in my latest release.

hopinxian commented on src/main/java/com/jacob/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,60 @@
+        String description = inputCommand.substring("deadline".length() + 1, breakpoint);
+        String dateTime = inputCommand.substring(breakpoint + 5);
+        Task theDeadline = new Deadline(description, dateTime);
+        taskList.add(theDeadline);
```
This block feels quite lengthy. Perhaps it would be better to add some line breaks for readability.

hopinxian commented on src/main/java/com/jacob/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+import main.java.com.jacob.duke.command.PrintListCommand;
+import main.java.com.jacob.duke.command.TodoCommand;
+
+public class Parser {
```
I believe there is quite some functions within other classes that would be better encapsulated within this class. Examples are detecting for invalid indexes, incomplete commands and so forth. I would suggest that Parser also have additional capabilities to identify invalid commands. I believe this would fall under the role of parsing the user input.

chuyouchia replied:
Yup agreed, thanks for the insight! I will probably refactor instead of using line breaks to improve readability

hopinxian reviewed (commented):
Overall, I like the assertions used to support your program. I think that is something I ought to learn. There are still some room for improvements. If you are using Intellij, I would suggest using the "Reformat code" function under the code tab. This would help you with the random white spaces that appear.

chuyouchia committed:
Add D-Notes as extension

chuyouchia committed:
Add D-Notes as extension

chuyouchia committed:
Update checkstyle imports errors

chuyouchia committed:
Refactor parser to deal with obvious incomplete commands

Place parser and storage into io package for better readability and
categorization

chuyouchia committed:
Update imports

chuyouchia committed:
Fix import statements

chuyouchia committed:
Merge pull request #3 from chuyouchia/branch-D-Notes

chuyouchia committed:
Add unit tests and exception handling for invalid commands.

chuyouchia committed:
Fix checkstyle errors

chuyouchia committed:
Move JavaFx related code into javafx package

chuyouchia committed:
Merge pull request #4 from chuyouchia/branch-extraexceptionhandling

Branch extra-exceptionhandling

chuyouchia committed:
Add Dark Theme and CSS stylings to Duke.

It is now called the fireside chat with the Mamba.

chuyouchia committed:
Add scroll pane resizing and dialog box resizing

chuyouchia committed:
Merge branch 'branch-A-BetterGUI'

chuyouchia committed:
Fix misspell in UI test due to added personality for chatbot

chuyouchia committed:
Add description for note related commands when starting up app

chuyouchia committed:
Add content for user guide into readme.md

chuyouchia committed:
Update Mamba interface linebreak in readme.md

chuyouchia committed:
Set theme jekyll-theme-hacker

chuyouchia committed:
Update README.md

chuyouchia committed:
Update README.md

chuyouchia committed:
Change help message format for date time format error in commands.

chuyouchia committed:
Fix delete command invalid index errors.

chuyouchia committed:
Update README.md

chuyouchia committed:
Update README.md

chuyouchia committed:
Add function for bye input to close window automatically.

chuyouchia committed:
Fix checkstyle error

chuyouchia committed:
Add credits to source where css file was adapted from.
